File: in_Java/0540 Single Element in a Sorted Array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        int i;
        int j;
        for(i = 0, j = 1; j < nums.length; i+=2, j+= 2){
            if(nums[i] != nums[j]) return nums[i];
        }
        return nums[i];
    }
}

File: in_Python/0540 Single Element in a Sorted Array.py
class Solution(object):
    def singleNonDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i,j = 0, len(nums)-1
        while i < j:
            m = i + (j-i)/2
            if m % 2 == 0:
                if nums[m] == nums[m+1]:
                    i = m + 2
                else:
                    j = m
            else:
                if nums[m] == nums[m-1]:
                    i = m + 1
                else:
                    j = m
        return nums[i]